packages/dashboard/src/customization/widgets/lineSymbol/component.test.tsx

```diff
@@ -0,0 +1,221 @@
+        color: 'black',
+        thickness: 5,
+        start: { x: 180, y: 175 },
+        end: { x: 20, y: 30 },
+      },
+    } as LineWidget;
+
+    it('should display anchors when the widget is selected', async () => {
+      // (useIsSelected as jest.Mock).mockReturnValue(true); // this makes it work, but we shouldn't need it
+      const { getByTestId, findByTestId } = render(<TestComponent widget={widget} isSelected={true} />);
+      expect(getByTestId('line-start-anchor')).toBeInTheDocument();
+      expect(getByTestId('line-end-anchor')).toBeInTheDocument();
+    });
+    it('should not display anchors when the widget is not selected', () => {
+      // (useIsSelected as jest.Mock).mockReturnValue(false);
+      const { queryByTestId } = render(<TestComponent widget={widget} isSelected={false} />);
+      expect(queryByTestId('line-start-anchor')).not.toBeInTheDocument();
+      expect(queryByTestId('line-end-anchor')).not.toBeInTheDocument();
+    });
+  });
+  describe('manipulating the widget', () => {
+    const beforeChangeWidget = {
+      ...mockWidgetPartial,
+      properties: {
+        lineStyle: 'solid',
+        color: 'black',
+        thickness: 5,
+        start: { x: 50, y: 50 },
+        end: { x: 100, y: 100 },
+      },
+    } as LineWidget;
+
+    const renderWithStore = (initialState: Partial<DashboardState>) => {
+      const store = configureDashboardStore(initialState);
+
+      const TestComponentWithStore = () => (
+        <Provider store={store}>
+          <DndProvider backend={HTML5Backend}>
+            <LineWidgetComponent {...(store.getState().dashboardConfiguration.widgets[0] as LineWidget)} />
+          </DndProvider>
+        </Provider>
+      );
+
+      return {
+        ...render(<TestComponentWithStore />),
+        store,
+      };
+    };
+
+    it('should set initial start and end points correctly on drag', async () => {
+      const initialState: Partial<DashboardState> = {
+        dashboardConfiguration: {
+          widgets: [beforeChangeWidget],
+          viewport: { duration: '5m' },
+        },
+        selectedWidgets: [beforeChangeWidget],
+      };
+
+      const { getByTestId, store } = renderWithStore(initialState);
+
+      const lineStartAnchor = getByTestId('line-start-anchor');
+      const startingPosition = {
+        clientX: 50,
+        clientY: 50,
+      };
+      const endingPosition = {
+        clientX: 100,
+        clientY: 120,
+      };
+
+      expect(store.getState().dashboardConfiguration.widgets[0].properties.start).toEqual({ x: 50, y: 50 });
+
+      act(() => {
+        fireEvent.dragStart(lineStartAnchor, startingPosition);
+        fireEvent.dragEnter(lineStartAnchor, endingPosition);
+        fireEvent.dragOver(lineStartAnchor, endingPosition);
+        fireEvent.drop(lineStartAnchor, endingPosition);
+        fireEvent.dragEnd(lineStartAnchor, endingPosition);
+      });
+
+      console.log(lineStartAnchor);
+
+      act(() => {
```

Comment:
take a look at the docs for act - https://github.com/threepointone/react-act-examples/blob/master/sync.md - might help you refactor this test.

Reply:
resolving since we will come back to this in a future pr